### ISSUE TITLE
I translated the software from GER to ENG. Any interested?

### ISSUE THREAD

**Issue by giobbino** [RetroNynjah/DELA-EPROMMER-II-27512-extension#1]
Hi,
not an issue, actually... 
I translated the software (both original and 512k versions) from german to english (at least I did my best), using Google and an hex editor. I wish to release it, are you interested to put it here?

**Comment by RetroNynjah** [comment]
I think it would be cool with an English version of the software. Maybe I should add the German standard version too now that Dirk has removed his repository from Github.

You can create a pull request and add the d64 with your versions to the software folder if you like. I'd be happy to merge them.
I should also create a readme for the software to explain the differences between the versions and how to use them.

**Comment by giobbino** [comment]
Fine. I tried to create a pull request, but I didn't understand how to do or the pull requests are disabled on your page.
As workaround, I create a fork of your project and I copied the Zip file with the translated software and manuals. 
You can get it here:
 
[https://github.com/giobbino/DELA-EPROMMER-II-27512-extension/tree/main/software](https://github.com/giobbino/DELA-EPROMMER-II-27512-extension/tree/main/software)

Since of course there's no need for a fork, please tell me when you're done and I will delete it.
Please fell free to improve my (rather poor) translation! :-)

**Comment by RetroNynjah** [comment]
Github can be quite confusing when you're not used to it.
To create a pull you must first create a fork where you make your changes and then create the pull request from your forked repository.

Anyway, I have added your translated software to this repository now so you can  remove the fork if you wish.
I think your translating will be easier to use than the german version for people that don't speak German.
Thanks for the files!

**Comment by giobbino** [comment]
Please take a look and tell me if it seems everything ok for you.

----------------------------------------------
btw I noticed the EPROM dump saved on .PRG by DELA eprommer is written in blocks of 2048 bytes that are in a wrong order if compared with the binary content of the EPROM (or any .bin image you can get from the web... zimmers.net, in example).

So I'm trying to write a small program to convert BIN to PRG and vice versa; this way, you can take a binary on the web, convert it as .prg, upload on a floppy disk (or sdcard) and use it directly from DELA eprommer.

If you have any idea or thought about this part, please let me know!

**Comment by RetroNynjah** [comment]
Sounds weird. I wasn't aware of that. I don't think I have dumped a ROM and used it for anything but I have programmed EPROMs using regular bin files like JiffyDOS. The only thing I did to the file was to add two extra bytes at the beginning of the file because the DELA software expects a load address and starts loading from the third byte into the adress of those first bytes.
You can specify manually where you want to load the file so the two bytes can be anything you want but they need to be there.

**Comment by giobbino** [comment]
Sounds weird to me too, but I did many tests because at first I didn't believe that. I programmed three EPROMs with my MiniPro (2764, 27128, 27256) and then I read the content with the DELA eprommer, and asked to save on disk (using an SD2IEC).
Then I extracted the three PRG (2764.PRG, 27128.PRG, 27256.PRG).

Taking the 2764 eprom as example....

I've seen that of course the PRG has 2 bytes at the beginning with the loading $1000 address written as 00 10. 
But the first byte of the bin file is written at the bottom of the .prg file (really weird too).

So, if you want to easily hex compare the two files, you only need to cut 1 byte from the .prg file and the content of the two files start to match from the 2nd byte until $07FF (2048 bytes). 

Then the 2nd block of 2048 bytes - $0800 to $0FFF on the PRG is at  $1800 to $1FFF on the BIN and vice versa.

I know that is really, really weird. I don't know why it happened, maybe it's my fault, I don't know.

I left here the content of my tests (original bin I used for the eprom and the .prg files I extracted using DirMaster from the .d64 image on the SDcard where DELA eprommer wrote the dump of the eproms).

https://github.com/giobbino/DELA-EPROMMER-II-27512-extension/blob/main/software/compare.zip

any idea?

**Comment by giobbino** [comment]
Some partial update: I wrote a first version of my software that should convert the BIN image to a DELA eprommer compatible PRG image.  

The test I did:
Starting from 8k and 16k .bin files, I've built 8k and 16k DELA images (.prg); I then put those PRGs into a .d64 image with the DELA eprommer software and I put the .d64 on an SDcard for my SD2IEC virtual disk drive.

I used those PRG to program two EPROMs (2764 and 27128) with the DELA programmer.

Finally, I compared the content of the two EPROM with the original BIN files using my MiniPro TL866 programmer: they are identical.  So it seems to me that a converter from BIN to PRG and vice versa is needed.

Note: I still need to solve some issues I'm having with the 27256 (it seems DELA is not correctly managing the load and save routines for 256kbit binaries).

**Comment by giobbino** [comment]
Update:

- I improved the 512k version of the software, adding the #6 menu item (27512) that was invisible in the German version.

- I wrote a software that manages the PRG and BIN files (for anybody that should experience the same issues I had). It's still a beta but fully working. It can transform PRG to BIN and BIN to PRG respecting the sequence that DELA uses for saving and loading the PRG to and from disk, adding or removing the loading address. This software is intended for anybody that should experience issues with PRG vs. BIN (for now it doesn't work with 27512 images; maybe I will add it in the future).

To leave things easier, I've built a repository with all the software, the manuals and my program in a single ZIP file.
In the README.MD I also added the links to this repository and the DELA Eprommer II Replica repository.
 
The link on my new repository is [https://github.com/giobbino/DELA-Eprommer-II-ENG-software-for-Commodore-64](https://github.com/giobbino/DELA-Eprommer-II-ENG-software-for-Commodore-64)

I also deleted the fork I made.
